src/logcumsumexp/problem_description.cpp

```diff
@@ -0,0 +1,209 @@
+                                                       const TensorDescriptor& outputDesc_,
+                                                       const TensorDescriptor& doutputDesc_,
+                                                       const TensorDescriptor& dinputDesc_,
+                                                       const int& dim_)
+    : ForwardProblemDescription(inputDesc_, outputDesc_, dim_),
+      doutputDesc(doutputDesc_),
+      dinputDesc(dinputDesc_)
+{
+    IsSameLength();
+    IsSameType();
+}
+
+bool BackwardProblemDescription::IsSameLength() const
+{
+    if(!ForwardProblemDescription::IsSameLength())
+        return false;
+    if(inputDesc.GetLengths() != dinputDesc.GetLengths())
+        MIOPEN_THROW(miopenStatusBadParm,
+                     "LogCumSumExp: Input and its Gradient tensor sizes do not match.");
+    if(outputDesc.GetLengths() != doutputDesc.GetLengths())
+        MIOPEN_THROW(miopenStatusBadParm,
+                     "LogCumSumExp: Output and its Gradient tensor sizes do not match.");
+    return true;
+}
+
+bool BackwardProblemDescription::IsSameType() const
+{
+    if(!ForwardProblemDescription::IsSameType())
+        return false;
+    if(inputDesc.GetType() != dinputDesc.GetType())
+        MIOPEN_THROW(miopenStatusBadParm,
+                     "LogCumSumExp: Input and its Gradient tensor type do not match.");
+    if(outputDesc.GetType() != doutputDesc.GetType())
+        MIOPEN_THROW(miopenStatusBadParm,
+                     "LogCumSumExp: Output and its Gradient tensor type do not match.");
+    return true;
+}
+
+bool BackwardProblemDescription::IsSameStride() const
+{
+    if(!ForwardProblemDescription::IsSameStride())
+        return false;
+    if(inputDesc.GetStrides() != dinputDesc.GetStrides())
+        return false;
+    if(outputDesc.GetStrides() != doutputDesc.GetStrides())
+        return false;
+    return true;
+}
+
+bool BackwardProblemDescription::IsAllPacked() const
+{
+    if(!ForwardProblemDescription::IsAllPacked())
+        return false;
+    if(!dinputDesc.IsPacked() || !doutputDesc.IsPacked())
+        return false;
+    return true;
+}
+
+bool BackwardProblemDescription::IsAllDimStride1() const
+{
+    if(!ForwardProblemDescription::IsAllDimStride1())
+        return false;
+    if(dinputDesc.GetStrides()[dim] != 1)
+        return false;
+    if(doutputDesc.GetStrides()[dim] != 1)
+        return false;
+    return true;
+}
+
+NetworkConfig ForwardProblemDescription::MakeNetworkConfig() const
+{
+    auto dtype      = inputDesc.GetType();
+    auto size       = inputDesc.GetElementSize();
+    auto inner_size = inputDesc.GetLengths()[dim];
+    auto outer_size = size / inner_size;
+
+    std::ostringstream ss;
+
+    ss << "logcumsumexp_fwd";
+    ss << "dtype" << dtype;
+    ss << "outer" << outer_size;
+    ss << "inner" << inner_size;
+    ss << "packed" << IsAllPacked();
+    ss << "dimstride1" << IsAllDimStride1();
```

Comment:
add `GetNumDims()` and `IsSameStride()` at here as it appears in `IsApplicable()`

Reply:
Resolved